src/pal/inc/rt/palrt.h

```diff
@@ -280,7 +280,7 @@ inline void *__cdecl operator new(size_t, void *_P)
 #define THIS_
 #define THIS                void
 
-#if _MSC_VER
+#if  defined(_MSC_VER) || defined(PAL_STDCPP_COMPAT)
```

Comment:
It is nothing to do with PAL_STDCPP_COMPAT. I meant to use a different check, but didn't end up doing it. Fixed.